api/src/pcapi/core/finance/api.py

```diff
@@ -1560,7 +1560,10 @@ def get_reimbursements_by_venue(
 ) -> typing.ValuesView:
     common_columns: tuple = (
         offerers_models.Venue.id.label("venue_id"),
-        sqla_func.coalesce(offerers_models.Venue.publicName, offerers_models.Venue.name).label("venue_name"),
+        sqla_func.coalesce(
+            sqla.case((offerers_models.Venue.publicName == "", None), else_=offerers_models.Venue.publicName),
+            offerers_models.Venue.name,
+        ).label("venue_name"),
```

Comment:
(détail / question de goût) Variante avec `nullif()` qui est peut-être un peu plus concise : 

```suggestion
        sqla_func.coalesce(sqla.nullif(offerers_models.Venue.publicName, ''), offerers_models.Venue.name).label("venue_name"),
```

Ceci dit, un rapide grep sur `Venue.publicName` montre qu'il y a d'autres endroits qui font la même erreur d'utilisation de `coalesce()`. Il faudrait donc une correction plus globale. Puisqu'il y a aussi de nombreux endroits où l'on écrit `venue.publicName or venue.name`, peut-être qu'on pourrait en profiter pour écrire une "hybrid_property" `best_name` (nom à discuter) qui renverrait `publicName` s'il n'est ni vide ni nul, ou `name` dans le cas contraire. On pourrait s'en servir ici et partout où l'on souhaite prendre `publicName` ou `name`.

Plus généralement, c'est un peu malheureux que ce le champ `Venue.publicName` soit NULLable puisque ça permet 2 valeurs vides : NULL et la chaîne vide. De manière générale, les colonnes TEXT devraient plutôt être NOT NULLable. Je ne sais pas si ça vaudrait la peine de modifier toutes les colonnes de notre base de données, mais ce serait un point à garder en tête (et à communiquer) pour nos futurs modèles ! :)

Reply:
Je suis d'accord pour la propriété "hybride" qui encapsulerait la logique de chercher le nom le plus approprié

J'aurais bien voulu utiliser la fonction sql `nullif`... si la méthode existait aussi dans `sqlalchemy`.
```python
In [1]: import sqlalchemy

In [2]: sqlalchemy.nullif
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 sqlalchemy.nullif

AttributeError: module 'sqlalchemy' has no attribute 'nullif'
```

Je t'avoue que je ne vois pas le rapport avec le problème qu'on a et le fait que `Venue.publicName` soit `nullable`. Dans tous les cas, des chaînes de caractères vides existent en base de donnée et sont sélectionnées par le `coalesce` qui ne passe à la valeur suivante qu'en cas de `null` et seulement `null`.

Reply:
J'ai fait une typo, la fonction est accessible via `sqlalchemy.func.nullif()`.

> Je t'avoue que je ne vois pas le rapport avec le problème qu'on a

Je ne comprends pas. Le rapport entre quoi et quoi ? (Et j'ai bien compris le bug.)